### PR TITLE
DAKIMAKURAS <3 (from pizzastation, bless)

### PR DESCRIPTION
**Pull request by ghost**
:cl: Kingofcarrotflowerspt1 and a tiny bit of Armhulen
add: Adds Dakimakuras, more colloquially known as "anime girl body pillows"
add: They take up a hand slot and can hold a few items. Many different girls to choose from!
/:cl:

I can finish this but I might want to hold off for a tiny bit to see if this is one of those insta-close prs.

I think they should be added because a lotta people (including me) wanted them added back when the first guy tried. Either way

image
and
image
and maybe a little bit
image

### CODE REVIEW

**Comment by evsey9** [comment]
Never clicked so fast

**Comment by Hatterhat** [comment]
holy SHIT @Armhulen you fucking madlad

**Comment by WJohn** [comment]
oh my god. this is amazing.

**Comment by TDcoolguy300** [comment]
I approve of this message.

**Comment by Armhulen** [comment]
/obj/item/weapon/storage/daki

code/_globalvars/lists/maintenance_loot.dm:271:error: /obj/item/weapon/storage/daki: undefined type path
??

**Comment by evsey9** [comment]
but is there a xenomorph bodypillow???

**Comment by Armhulen** [comment]
i could sprite a xenomorph bodypillow

**Comment by juanototo2** [comment]
Do it so wjohn will speedmerge

**Comment by evsey9** [comment]
default

**Comment by Artemiuz28** [comment]
DAKIMURAS? THE FUCK IS THIS? PLEASE FIX THE NAME OF THIS PR IMMEDIATELY OH GOD

**Comment by juanototo2** [comment]
Dakimakuras are another creation from japan caused due to waifu proliferation. Basically anime character body pillows.

**Comment by Armhulen** [comment]
will that do sir

**Comment by SgtxRootbeer** [comment]
@Armhulen please let me tape photographs of people to the dakimuras, to change the name and image!

**Comment by Armhulen** [comment]
:heart: too :heart: much :heart: work :heart: 

i'm already porting this ancient code

File: code/game/objects/items/dakis.dm [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ 	icon_state = "daki_base"
+ 	slot_flags = SLOT_BACK
+ 	storage_slots = 3
+ 	w_class = 4
+ 	max_w_class = 3
+ 	max_combined_w_class = 21
+ 	var/cooldowntime = 20
+ 	var/static/list/dakimakura_options = list("Callie","Casca","Chaika","Elisabeth","Foxy Grandpa","Haruko","Holo","Ian","Jolyne","Kurisu","Marie","Mugi","Nar'Sie","Patchouli","Plutia","Rei","Reisen","Naga","Squid","Squigly","Tomoko","Toriel","Umaru","Yaranaika","Yoko") //Kurisu is the ideal girl." - Me, Logos.
+
+ /obj/item/weapon/storage/daki/attack_self(mob/living/user)
```
Comment (ShizCalev) [inline code/game/objects/items/dakis.dm]: ITEM/WEAPON REEEEEEEE

Comment (Armhulen) [reply code/game/objects/items/dakis.dm]: i already fixed this let me sync

**Review by deathride58 — COMMENTED** [review]
Where the fuck is daki.dmi

Check your code before you fuck your code

**Comment by Repukan** [comment]
Slippery slope

Make gachimuchi dakis and I'll retract my criticisms

**Comment by WJohn** [comment]
you know at some with the whole slippery slope, once you're down far enough why not just simply *jump off?*

**Comment by Repukan** [comment]
Why not climb out.

**Comment by KorPhaeron** [comment]
I was gonna say you need to post screenshots but then I realized you forgot the dmi entirely

**Comment by Armhulen** [comment]
sprites in T-2 minutes

**Comment by Iamgoofball** [comment]
Gachimuchi dakis or bust.

**Comment by Armhulen** [comment]
i guess i could sprite that

File: code/_globalvars/lists/maintenance_loot.dm [hunk 2]
```diff
@@ -104,5 +104,6 @@ GLOBAL_LIST_INIT(maintenance_loot, list(
 	/obj/item/storage/secure/briefcase = 3,
 	/obj/item/storage/toolbox/artistic = 2,
 	/obj/item/toy/eightball = 1,
+	/obj/item/weapon/storage/daki = 3,
```
Comment (ShizCalev) [inline code/_globalvars/lists/maintenance_loot.dm]: here too.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by Armhulen** [comment]
tgstation.dmb - 0 errors, 0 warnings (10/5/17 7:53 pm)

File: code/_globalvars/lists/maintenance_loot.dm (hunk 2 above)
Comment (Armhulen) [reply code/_globalvars/lists/maintenance_loot.dm]: already fixed again!

File: tgstation.dme [hunk 3]
```diff
@@ -745,6 +745,7 @@
 #include "code\game\objects\items\cosmetics.dm"
 #include "code\game\objects\items\courtroom.dm"
 #include "code\game\objects\items\crayons.dm"
+#include "code\game\objects\items\dakis.dm"
```
Comment (ShizCalev) [inline tgstation.dme]: code/game/objects/items/storage/dakis.dm???

Comment (Armhulen) [reply tgstation.dme]: they store items, what's wrong?

Comment (Armhulen) [reply tgstation.dme]: >not hiding bear traps in your waifu to protect her against brainless orc assistants

??

Comment (ShizCalev) [reply tgstation.dme]: wea8ol7
8wbndtc
zzcjoyo

**Comment by Qustinnus** [comment]
help my armhullen is broken they are making bad prs on my codebase. anyone know how to reset them?? thank you

File: tgstation.dme (hunk 3 above)
Comment (Armhulen) [reply tgstation.dme]: shiiiiii how do i fix this one second

Comment (Armhulen) [reply tgstation.dme]: FIXED

**Comment by Armhulen** [comment]
BAD?

**Comment by The-clown-known-as-Pepe** [comment]
👎

**Comment by optimumtact** [comment]
WHO AWAKENS ME FROM MY SLUMBER

File: code/game/objects/items/storage/dakis.dm [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+				playsound(src, "rustle", 50, 1, -5)
+			if(INTENT_GRAB)
+				user.visible_message("<span class='warning'>[user] holds the [name]!</span>")
+				playsound(src, 'sound/items/bikehorn.ogg', 50, 1)
+			if(INTENT_HARM)
+				user.visible_message("<span class='danger'>[user] punches the [name]!</span>")
+				playsound(src, 'sound/effects/shieldbash.ogg', 50, 1)
+		user.changeNext_move(CLICK_CD_MELEE)
+
+ ////////////////////////////
```
Comment (ShizCalev) [inline code/game/objects/items/storage/dakis.dm]: Remove these slashes, it'll cause new line bs whenever someone edits the file since it's at the end.

**Comment by Armhulen** [comment]
aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaah

**Comment by Armhulen** [comment]
@optimumtact 



she's waiting for you, oranges.

**Review by ShizCalev — CHANGES_REQUESTED** [review]
I spent like 6 hours cleaning out icons/mob/inhands/items_lefthand.dmi &  icons/mob/inhands/items_righthand.dmi

Move the 52 inhand sprites to a new files in icons\mob\inhands\misc instead please.

**Comment by Armhulen** [comment]
can do, my friend

**Comment by Armhulen** [comment]
but, i'm about to go on a vacation so this might be a little bit

**Comment by ShizCalev** [comment]
Personally, I'm not a fan of the quality of the pillow's base sprite. Looks like something from 2011.

**Comment by Armhulen** [comment]
Okay, I moved everything over!

**Comment by Armhulen** [comment]
image
IT WORKS LADS

File: code/game/objects/items/storage/dakis.dm [hunk 5]
```diff
@@ -0,0 +1,44 @@
+//////////////////////////////////
+ //dakimakuras
+ //////////////////////////////////
+
+/obj/item/storage/daki
+	name = "dakimakura"
+	desc = "A large pillow depicting a girl in a compromising position. Featuring as many dimensions as you."
+	icon = 'icons/obj/daki.dmi'
+	icon_state = "daki_base"
```
Comment (ShizCalev) [inline code/game/objects/items/storage/dakis.dm]: Need to define the inhand location.

lefthand_file = 'icons/mob/inhands/misc/daki_lefthand.dmi'
righthand_file = 'icons/mob/inhands/misc/daki_righthand.dmi'

**Review by ShizCalev — DISMISSED** [review]

File: code/game/objects/items/storage/dakis.dm (hunk 5 above)
Comment (Armhulen) [reply code/game/objects/items/storage/dakis.dm]: Sure I'll do it in the morning though

**Comment by Arianya** [comment]
HoP needs to start with a Ian daki to match his other Ian merch

**Comment by lzimann** [comment]
This won't be merged and probably closed at 24h.

**Comment by GuyonBroadway** [comment]
Last time these were on hippie, I gathered a few blank ones as a Mime, I slipped, cuffed and gagged some random people from the hallway, dragged them to a secluded part of maint and made a daki using their name.

I then beat the shit out of said daki for 2 mins, dragged the victim back to the hallway and left them standing there. 

I did this two or three times before sec executed me on the spot.

**Comment by Armhulen** [comment]
Okay this is 100% fully working and everything!

File: code/game/objects/items/storage/dakis.dm [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+	var/cooldowntime = 20
+	var/static/list/dakimakura_options = list("Callie","Casca","Chaika","Elisabeth","Foxy Grandpa","Haruko","Holo","Ian","Jolyne","Kurisu","Marie","Mugi","Nar'Sie","Patchouli","Plutia","Rei","Reisen","Naga","Squid","Squigly","Tomoko","Toriel","Umaru","Yaranaika","Yoko") //Kurisu is the ideal girl." - Me, Logos. //how could someone make something so good then say some retarded shit like this
+
+/obj/item/storage/daki/attack_self(mob/living/user)
+	var/body_choice
+	var/custom_name
+
+	if(icon_state == "daki_base")
+		body_choice = input("Pick a body.") in dakimakura_options
+		icon_state = "daki_[body_choice]"
```
Comment (Cyberboss) [inline code/game/objects/items/storage/dakis.dm]: needs sanity checks after input

File: code/game/objects/items/storage/dakis.dm [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+		custom_name = stripped_input(user, "What's her name?")
+		if(length(custom_name) > MAX_NAME_LEN)
```
Comment (Cyberboss) [inline code/game/objects/items/storage/dakis.dm]: ditto

File: code/game/objects/items/storage/dakis.dm [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+			to_chat(user,"<span class='danger'>Name is too long!</span>")
+			return FALSE
+		if(custom_name)
+			name = custom_name
+			desc = "A large pillow depicting [custom_name] in a compromising position. Featuring as many dimensions as you."
+	else
+		switch(user.a_intent)
+			if(INTENT_HELP)
+				user.visible_message("<span class='notice'>[user] hugs the [name].</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/storage/dakis.dm]: name -> src, remove `the`

File: code/game/objects/items/storage/dakis.dm [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+				playsound(src, "rustle", 50, 1, -5)
+			if(INTENT_DISARM)
+				user.visible_message("<span class='notice'>[user] kisses the [name].</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/storage/dakis.dm]: name -> src, remove `the`

File: code/game/objects/items/storage/dakis.dm [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+				playsound(src, "rustle", 50, 1, -5)
+			if(INTENT_GRAB)
+				user.visible_message("<span class='warning'>[user] holds the [name]!</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/storage/dakis.dm]: name -> src, remove `the`

File: code/game/objects/items/storage/dakis.dm [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+				playsound(src, 'sound/items/bikehorn.ogg', 50, 1)
+			if(INTENT_HARM)
+				user.visible_message("<span class='danger'>[user] punches the [name]!</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/storage/dakis.dm]: name -> src, remove `the`

File: code/game/objects/items/storage/dakis.dm (hunk 8 above)
Comment (Cyberboss) [reply code/game/objects/items/storage/dakis.dm]: Needs second parameter

File: code/game/objects/items/storage/dakis.dm (hunk 9 above)
Comment (Cyberboss) [reply code/game/objects/items/storage/dakis.dm]: Needs second parameter

File: code/game/objects/items/storage/dakis.dm (hunk 10 above)
Comment (Cyberboss) [reply code/game/objects/items/storage/dakis.dm]: Needs second parameter

File: code/game/objects/items/storage/dakis.dm (hunk 11 above)
Comment (Cyberboss) [reply code/game/objects/items/storage/dakis.dm]: Needs second parameter

File: code/game/objects/items/storage/dakis.dm [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+
+	if(icon_state == "daki_base")
+		body_choice = input("Pick a body.") in dakimakura_options
+		icon_state = "daki_[body_choice]"
+		custom_name = stripped_input(user, "What's her name?")
+		if(length(custom_name) > MAX_NAME_LEN)
+			to_chat(user,"<span class='danger'>Name is too long!</span>")
+			return FALSE
+		if(custom_name)
+			name = custom_name
```
Comment (Cyberboss) [inline code/game/objects/items/storage/dakis.dm]: name = "\proper [custom_name]"

**Review by Cyberboss — DISMISSED** [review]

**Comment by Armhulen** [comment]
dammit okay one second!

File: code/game/objects/items/storage/dakis.dm (hunk 6 above)
Comment (Armhulen) [reply code/game/objects/items/storage/dakis.dm]: ?

**Comment by ExcessiveUseOfCobblestone** [comment]
Is this your equivalent to my flashsuit pr

File: code/game/objects/items/storage/dakis.dm (hunk 6 above)
Comment (Qustinnus) [reply code/game/objects/items/storage/dakis.dm]: a sanity check is checking if something is still viable, it prevents runtimes.

File: code/game/objects/items/storage/dakis.dm [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+	desc = "A large pillow depicting a girl in a compromising position. Featuring as many dimensions as you."
+	icon = 'icons/obj/daki.dmi'
+	icon_state = "daki_base"
+	lefthand_file = "icons/mob/inhands/misc/daki_lefthand.dmi"
+	righthand_file = "icons/mob/inhands/misc/daki_righthand.dmi"
+	slot_flags = SLOT_BACK
+	storage_slots = 3
+	w_class = WEIGHT_CLASS_NORMAL
+	var/cooldowntime = 20
+	var/static/list/dakimakura_options = list("Callie","Casca","Chaika","Elisabeth","Foxy Grandpa","Haruko","Holo","Ian","Jolyne","Kurisu","Marie","Mugi","Nar'Sie","Patchouli","Plutia","Rei","Reisen","Naga","Squid","Squigly","Tomoko","Toriel","Umaru","Yaranaika","Yoko") //Kurisu is the ideal girl." - Me, Logos. //how could someone make something so good then say some retarded shit like this
```
Comment (KorPhaeron) [inline code/game/objects/items/storage/dakis.dm]: These all need to be SS13 related and you need way better sprites

**Review by imtakingabreakdontatme — CHANGES_REQUESTED** [review]

File: code/game/objects/items/storage/dakis.dm (hunk 13 above)
Comment (Armhulen) [reply code/game/objects/items/storage/dakis.dm]: i am willing to rework all of these but i can't really that right now as i'm leaving for a vacation very very very soon

Comment (Armhulen) [reply code/game/objects/items/storage/dakis.dm]: so i'd rather get this in then when i'm back edit them to be thematic!

**Comment by DoctoreMad** [comment]
​>From hippie
​>Credit where it's due
Fucking new_comers_ , stealing our legitimate credit. But I suppose you don't **need** to know where it actually came form.

**Comment by Armhulen** [comment]
i'll rename it for you

**Comment by praisenarsie** [comment]
tswing1

#FROGGATE

File: code/game/objects/items/storage/dakis.dm (hunk 13 above)
Comment (KorPhaeron) [reply code/game/objects/items/storage/dakis.dm]: Maybe whichever spriter you bribe can PR your code for you as well then, if it was merely bad sprites I'd go with it but this is a pretty big violation of the reference policy etc

**Comment by Armhulen** [comment]
The name thing isn't showing up I'll fix it asap!

**Comment by Xhuis** [comment]
*There's a point where we need to stop and we've passed it **BUT LET'S KEEP GOING AND SEE WHAT HAPPENS***

File: code/game/objects/items/storage/dakis.dm (hunk 13 above)
Comment (Armhulen) [reply code/game/objects/items/storage/dakis.dm]: you're lucky i care about dakis more than i do my trip, i'll swap them out either today or tomorrow!

File: code/game/objects/items/storage/dakis.dm (hunk 5 above)
Comment (Armhulen) [reply code/game/objects/items/storage/dakis.dm]: Added you can close this I guess

**Comment by SyruseEln** [comment]
I really want this. Like, a LOT!

**Comment by mattroks101** [comment]
Good PR.

**Comment by Hatterhat** [comment]
BRING IT BACK

**Comment by lzimann** [comment]
since the script didn't post this:
I really don't think we should be encouraging this kind of content. Plus, this has been denied(with a merge/fast revert merge) before, so makes no sense to add this time. So, unless kor for some probably awful reason wants this, i'll just close and we can all move on with our lives.

**Comment by Mercenaryblue** [comment]
> you'll never cuddle with your Nar'Sie pillow.
Why even spacemen

**Comment by FantasticFwoosh** [comment]
Its a bit too late for such a sentiment of 'kinds of content' @lzimann big server business relies on selling your integrity to appeal to the _people_ just so they pitch up. Least until your obstruction means anything to the persons who push this on at-least a annual basis.

- Can i just say i don't know for reference a single name presented on that list of overlay "girls" or who they are meant to represent. If you could calmly attach a short description in the chat id uh, acknowledge it, maybe not quite be thankful for it.

**Comment by ShizCalev** [comment]
@FantasticFwoosh 

"Callie" https://splatoonwiki.org/wiki/Callie
"Casca" http://berserk.wikia.com/wiki/Casca
"Chaika" http://hitsugimenochaika.wikia.com/wiki/Chaika_Trabant
"Elisabeth" //not sure
"Haruko" http://flcl.wikia.com/wiki/Haruko_Haruhara
"Holo" http://spiceandwolf.wikia.com/wiki/Holo
"Jolyne" http://jojo.wikia.com/wiki/Jolyne_Cujoh
"Kurisu" http://steins-gate.wikia.com/wiki/Kurisu_Makise
"Marie" //not sure
"Mugi" http://k-on.wikia.com/wiki/Tsumugi_Kotobuki
"Patchouli" https://en.touhouwiki.net/wiki/Patchouli_Knowledge
"Plutia" http://hyperdimensionneptunia.wikia.com/wiki/Plutia
"Rei" https://en.wikipedia.org/wiki/Rei_Ayanami
"Reisen" https://en.touhouwiki.net/wiki/Reisen_Udongein_Inaba
"Naga" //standard naga. don't think this was an issue.
"Squid" //not sure
"Squigly" http://skullgirls.wikia.com/wiki/Squigly
"Tomoko" http://watamote.wikia.com/wiki/Tomoko_Kuroki
"Toriel" http://undertale.wikia.com/wiki/Toriel
"Umaru" http://himoutoumaru-chan.wikia.com/wiki/Umaru_Doma
"Yoko" http://gurrenlagann.wikia.com/wiki/Yoko_Littner

**Comment by Iamgoofball** [comment]
Holy exterminatus

**Comment by Repukan** [comment]
Marie is from Splatoon too

**Comment by FantasticFwoosh** [comment]
Im so glad i don't play here anymore.  

Splatoon is fictional, but it (and other underage models) should be removed in order to comply with law regulations, because not every country in the world likes putting kids or even abstract representations of cephalopod kids on dubious items like this outside of Japan.

- People under 18 aren't even allowed onto the server as per rules laid out to MrStonedOne, we still have a standard to uphold (that it doesn't violate US law) so they don't pull the plug.

Honestly this is the most gentle way I can say this without being accusatory at the top of my lungs.

**Comment by Bawhoppen** [comment]
Goddamnit Fwoosh

**Comment by FantasticFwoosh** [comment]
Furthermore putting photos that are ERP innapropriate onto  automatically lets you facilitate a **rule 8** ban (though a circumnavigating of the rules by saying its a coded feature! will still be met with a **rule 7** by using coders as a strawman to let you do it.)

Hence begrudgingly id prefer photos of IC station events & people to creepy underage weeb references submitted by neckbeards (nothing personal Armhulen)

**Comment by Qustinnus** [comment]
someone got their body pillow stolen as a kid i see

**Comment by Bawhoppen** [comment]
Goddamnit Fwoosh just stop posting

**Comment by sawrge** [comment]
Lmfao holy shit

**Comment by sawrge** [comment]
I don't know if Lzimann writing a script to auto close this after 24 hours or Fwoosh actually writing any of the insanely retarded shit he wrote is funnier. What do you guys think

**Comment by lzimann** [comment]
yea, 10 secs wasted writing a script that i'll never get back

**Comment by optimumtact** [comment]
This is assigned to me, not you @lzimann so I would appreciate if you woudl step off and keep your grubby hands to yourself

**Comment by optimumtact** [comment]
also don't get into a script war or I'll have @KorPhaeron revoke your pr access

**Comment by optimumtact** [comment]
fuck the comment quality has gone down hill recently

**Comment by optimumtact** [comment]
actually I'm just going to keep this preemptively locked until I can chat with kor

**Comment by lzimann** [comment]
nah, fuck you, why do you suddenly care about this but ignored for 3 days?

**Comment by Cyberboss** [comment]
I wonder what this button does...

**Comment by Cyberboss** [comment]
4:53:58 PM armhulennn Cyberboss or thot_slayer could you post that i'm working on sprites for the dakis right now since i can't (locked)

**Comment by Iamgoofball** [comment]
gachimuchi or bust
i will not be divided

**Comment by Armhulen** [comment]
I'll pr more sprites once this is in, I just want this here to start
speaking of:
custom_name bug is fixed
all sprites are in
all references to not-spacestation are out!

**Comment by Hatterhat** [comment]
|  ||
|| |_

**Review by ShizCalev — CHANGES_REQUESTED** [review]
Update the inhands with the new sprites too please.

**Comment by Armhulen** [comment]
oh, of course. gimmie a bit and I will!

**Comment by WJohn** [comment]
Having seen the sprites, these are total abominations. Holy shit they're worse than the old whetstone sprites by like a mile. They're worse than oldstation sprites.

**Comment by KorPhaeron** [comment]
>nah, fuck you, why do you suddenly care about this but ignored for 3 days?

I don't think three days is a very long time though? It isn't sane or reasonable to expect volunteers to have a sub 24 hour response time 365 days a year

**Comment by Armhulen** [comment]
@WJohn I have an idea for improved sprites, having simplemob icons on the pillow

**Pull request closed without merging**